src/directives/onLongPress.ts

```diff
@@ -0,0 +1,137 @@
+import {ElementPart, nothing} from 'lit';
+import {AsyncDirective, directive, PartType} from 'lit/async-directive.js';
+import type {
+  DirectiveParameters,
+  DirectiveClass,
+  DirectiveResult,
+  PartInfo
+} from 'lit/async-directive.js';
+
+type LongPressCallback = (event?: PointerEvent) => void;
+
+const DEFAULT_LONG_PRESS_TIMEOUT_MS = 1000;
+
+class LongPressDirective extends AsyncDirective {
+  /** Element of the directive. */
+  #element!: Element;
+
+  /**
+   * Long press timeout.
+   * This timeout initiates when the pointer is pressed on the
+   * element. It calls user's callback unless cancel events occur
+   * before time's out.
+   */
+  #longPressTimeout?: NodeJS.Timeout;
+
+  /** Time before the timeout runs out. */
+  #longPressTimeoutMs?: number | undefined;
+
+  /** User-defined callback for long-press event */
+  #longPressCallback?: LongPressCallback;
+
+  constructor(partInfo: PartInfo) {
+    super(partInfo);
+    if (partInfo.type !== PartType.ELEMENT) {
+      throw new Error(
+        "Can't bind `onLongPress` directive to anything " +
+          'that is not an element.'
+      );
+    }
+    this.#updateElement((partInfo as ElementPart).element);
+  }
+
+  /** @inheritdoc */
+  render(
+    callback: LongPressCallback,
+    callbackTimeoutMs = DEFAULT_LONG_PRESS_TIMEOUT_MS
+  ) {
+    this.#longPressCallback = callback;
+    this.#longPressTimeoutMs = callbackTimeoutMs;
+
+    return nothing;
+  }
+
+  /** @inheritdoc */
+  public override update(
+    part: ElementPart,
+    [callback, callbackTimeoutMs]: DirectiveParameters<this>
+  ): void {
+    this.render(callback, callbackTimeoutMs);
+    if (part.element !== this.#element) {
+      this.#updateElement(part.element);
+    }
+  }
+
+  #updateElement(element: Element) {
+    // Detach events from previous element
+    if (this.#element !== undefined) {
+      this.#detachEvents();
+    }
+    this.#element = element;
+    this.#attachEvents();
+  }
+
+  #attachEvents() {
```

Comment:
once you remove the non-null assertion on `element!` at the top of this file, this method and `detachEvents` should probably just take an element as a parameter:

```ts
#attachEvents(node: Element): void {
  node.addEventListener(...);
  // etc...
}
```